tests/5.0/teams_loop/test_target_teams_loop_allocate.c

```diff
@@ -0,0 +1,259 @@
+  omp_large_cap_mem_alloc allocator to allocate an array.
+*/
+int testTargetTeamsAllocateArrayLargeCapAlloc() {
+  int a[N], b[N], c[N], local[N];
+  int errors = 0;
+  // Data Inititalize
+  for (int i = 0; i < N; i++) {
+    a[i] = 2*i;  // Even
+    b[i] = 2*i + 1;  // Odd
+    local[i] = 0;
+  }
+  // Execute on target
+#pragma omp target teams loop map(to: a[0:N], b[0:N]) map(from: c[0:N])\
+        uses_allocators(omp_large_cap_mem_alloc)\
+        allocate(omp_large_cap_mem_alloc: local) private(local)
+  for (int i = 0; i < N; i++) {
+    local[i] = a[i] + b[i];
+    c[i] = local[i];
+  }
+  // Validate
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET_VERBOSE(errors, c[i] != a[i] + b[i]);
+    OMPVV_TEST_AND_SET_VERBOSE(errors, local[i] != 0);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate use of allocate clause with
+  omp_const_mem_alloc allocator to allocate an variable.
+*/
+int testTargetTeamsAllocateConstAlloc() {
+  int a[N], b[N];
+  const int local = 3;
+  int errors = 0;
+  // Data Inititalize
+  for (int i = 0; i < N; i++) {
+    a[i] = 2*i;  // Even
+    b[i] = 0;  // Odd
+  }
+  // Execute on target
+#pragma omp target teams loop map(to: a[0:N]) map(from: b[0:N])\
+        firstprivate(local) uses_allocators(omp_const_mem_alloc)\
+         allocate(omp_const_mem_alloc: local)
+  for (int i = 0; i < N; i++) {
+    b[i] = a[i] + local;
+  }
+  // Validate
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET_VERBOSE(errors, b[i] != a[i] + local);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate use of allocate clause with
+  omp_high_bw_mem_alloc allocator to allocate an array.
+*/
+int testTargetTeamsAllocateArrayHighBWAlloc() {
+  int a[N], b[N], c[N], local[N];
+  int errors = 0;
+  // Data Inititalize
+  for (int i = 0; i < N; i++) {
+    a[i] = 2*i;  // Even
+    b[i] = 2*i + 1;  // Odd
+    local[i] = 0;
+  }
+  // Execute on target
+#pragma omp target teams loop map(to: a[0:N], b[0:N]) map(from: c[0:N])\
+        uses_allocators(omp_high_bw_mem_alloc)\
+        allocate(omp_high_bw_mem_alloc: local) private(local)
+  for (int i = 0; i < N; i++) {
+    local[i] = a[i] + b[i];
+    c[i] = local[i];
+  }
+  // Validate
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET_VERBOSE(errors, c[i] != a[i] + b[i]);
+    OMPVV_TEST_AND_SET_VERBOSE(errors, local[i] != 0);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate use of allocate clause with
+  omp_low_lat_mem_alloc allocator to allocate an array.
+*/
+int testTargetTeamsAllocateArrayLowLatAlloc() {
+  int a[N], b[N], c[N], local[N];
+  int errors = 0;
+  // Data Inititalize
+  for (int i = 0; i < N; i++) {
+    a[i] = 2*i;  // Even
+    b[i] = 2*i + 1;  // Odd
+    local[i] = 0;
+  }
+  // Execute on target
+#pragma omp target teams loop map(to: a[0:N], b[0:N]) map(from: c[0:N])\
+        uses_allocators(omp_low_lat_mem_alloc)\
+        allocate(omp_low_lat_mem_alloc: local) private(local)
```

Comment:
The error @tob2 mentioned still exists here (`error: 'local' specified in 'allocate' clause but not in an explicit privatization clause`); one possible solution is to use `firstprivate(local)` instead of `private(local)`, but using the `firstprivate` clause will incur unnecessary initialization of the firstprivate variable, `local`.

Reply:
Replaced "private(local)" with  "firstprivate(local)" as suggested...